src/pages/profile.tsx

```diff
@@ -437,16 +471,29 @@ const Profile: NextPage = () => {
                 <div className="my-4">
                   <div className="border-l-2 border-l-black">
                     {daysOfWeek.map((day, index) => (
-                      <Checkbox
+                      <Controller
                         key={day + index.toString()}
-                        sx={{
-                          input: { width: 1, height: 1 },
-                          padding: 0,
-                        }}
-                        {...register(`daysWorking.${index}`)}
-                        checkedIcon={<DayBox day={day} isSelected={true} />}
-                        icon={<DayBox day={day} isSelected={false} />}
-                        defaultChecked={false}
+                        name={`daysWorking.${index}`}
+                        control={control}
+                        render={({
+                          field: { onChange, value },
+                          formState: { defaultValues },
+                        }) => (
+                          <Checkbox
+                            key={day + index.toString()}
+                            sx={{
+                              input: { width: 1, height: 1 },
+                              padding: 0,
+                            }}
+                            checked={value}
+                            onChange={onChange}
+                            checkedIcon={<DayBox day={day} isSelected={true} />}
```

Comment:
what's the difference between the checked field and checkedIcon?

Reply:
@lukejianu